steel/src/contract.rs

```diff
@@ -286,13 +289,15 @@ where
     ///
     /// [EvmEnv]: crate::EvmEnv
     pub fn call(self) -> C::Return {
+        // safe unwrap: env is never returned without a DB
         let state_db = self.env.db.as_ref().unwrap();
         let mut evm = new_evm::<_, H>(
             WrapStateDb::new(state_db),
             self.env.cfg_env.clone(),
             self.env.header.inner(),
         );
-        self.tx.transact(&mut evm).unwrap()
+        // typically, guest functions should panic instead of returning an error
+        self.tx.transact(&mut evm).expect("call failed")
```

Comment:
```suggestion
        // typically, guest functions should panic instead of returning an error
        self.tx.transact(&mut evm)
```

Reply:
OK, that was my impression that we often prefer panicking over errors in the guest such as in `env::read()` or other `env`s.

Reply:
I have changed that in 17b4550b97a91037dbef2d1cadc171fd6b5f5594

Reply:
It depends, ya. I'd say as a rule of thumb that guest panicking when the error could only have been caused by host misbehavior, ya (e.g. in the `k256` patch, I added a panic if the host did not normalize the result). In other cases, it's a judgement call. I lean more towards not panicking anywhere except the root of the guest (i.e. in `main.rs`)